### ISSUE TITLE
After correcting flagged issue in DOM in Firefox Mac, cannot get axe-core to re-flag the issue on page reload

### ISSUE THREAD

**Issue by CharlesBelov** [dequelabs/axe-core#3971]
### Product

axe-core

### Product Version

extension: v4.49.0 axe-core 4.6.3 

### Latest Version

- [X] I have tested the issue with the latest version of the product

#### Expectation
If I am visiting a webpage in Firefox Mac and axe-core reports problems on that page, then I use Firefox's inspector to attempt a correction, I can then ask axe-core to reinspect the page and verify that the new error count is zero. If I subsequently reload the page, then re-run axe-core, I expect it to re-find the errors.

#### Actual
If I am visiting a webpage in Firefox and axe-core reports problems on that page, then I use Firefox's inspector to attempt a correction, I can then ask axe-core to reinspect the page and verify that the new error count is zero. If I subsequently reload the page, then re-run axe-core, the error count remains at zero. Clearing history does not correct the issue. I have to delete the Firefox profile and readd it under a new name to get the errors to show again.

This appears to be an intermittent problem. It happened the first time I tried this but not the second; the second time it worked correctly.

#### How to Reproduce
The [SFMTA.com homepage](https://www.sfmta.com) currently produces this issue. However, I plan to have this issue corrected and it may not remain as an error. Here is the code which is subjected to by axe-core:

axe-core correctly reports a violation for 1.4.3 for class `.compact-form-label`:

`<label class="control-label compact-form-label" for="edit-text"><span class="search-icon"></span> Search SFMTA...</label>`

Element has insufficient color contrast of 4.44 (foreground color: #75787b, background color: #ffffff, font size: 10.8pt (14.4px), font weight: bold). Expected contrast ratio of 4.5:1

axe-core correctly reports a violation of 2.4.4 and 4.1.2 for class .btn-search:

`<a href="/search" class="btn-search"><span class="search-icon"></span></a>`

Actual steps:
1. Launch Firefox with the axe-core extension
2. Go to https://www.sfmta.com (again, this may not reproduce the error in the future because we plan to fix the page)
3. Right click Careers in the top navigation bar
4. Choose inspect (note: I am using a floating development tools window)
5. Go to the axe DevTools tab
6. Click scan all of my page
7. Note the two errors above
8. Go to the Inspector tab
9. Option click the expand arrow for the element having ID equal to navigation-right
10. Scroll down and locate the a tag with class equals btn-search and click it
11. Insert the following attribute into the a tag:  title="Search"
12. Continue scrolling down in the inspector
13. Locate the label tag with the class including compact-form-label and click it
14. In the styles column, locate the color property and change the value from #75787b to #232325
15. Return to the axe DevTools tab
16. Click rerun scan

Actual and expected results: I am now getting zero errors.

17. Click on the browser window
18. Click the refresh icon to reload the page
19. Right click Careers in the top navigation bar
20. Choose inspect (note: I am using a floating development tools window)
21. Scroll down and locate the a tag with class equals btn-search and click it. Note that it no longer contains the title attribute that I added because I refreshed the page
22. Continue scrolling down in the inspector
23. Locate the label tag with the class including compact-form-label and click it
24. In the styles column, locate the color property. Note that it is once again #75787b because I refreshed the page
25. Return to the axe DevTools tab
26. Click rerun scan

Expected and second try actual results: 2 errors are found

First try actual result: no errors found

#### Additional context
Any thing else we should know about the issue?

Meta steps:
1. Launched existing Firefox profile containing axe-core
2. Check whether axe-core was running the latest version and got the update to the current version
3. Went through the many steps above, which resulted in the reported problem, which I was unable to clear through reloads or clearing history.
4. Deleted the Firefox profile
5. Created a different Firefox profile with a new name
6. Set the various Firefox settings, which I cannot guarantee are the same as the previous settings but which generally involve turning off suggestions and clearing history on accent
7. Re-went through the many steps above, and was this time unable to re-create the problem

**Comment by CharlesBelov** [comment]
As of April 5, 2023, the errors are no longer being detected on this page. We have not updated the code, and I am still on axe-core 4.6.3. I have not launched this particular Firefox profile since I reported the issue.

However, if I create a fresh Firefox profile, the errors are detected, so I think this is a recurrence of the issue I reported above, happening across separate launches of the Firefox profile.
